### DISCUSSION TITLE
Treat Java's `of` convention the same Scala's `apply`

### DISCUSSION THREAD

**Discussion by dhinojosa** [scala/scala3#17872] [Feature Requests]
It would be great to leave out any calls to `of` in the Java API much like we do with `apply`. 

Whereas

```scala
Foo.apply(13)
```

is the same as 
```scala
Foo(13)
```

it would be great to include the Java convention

```
LocalDate.of(2021, 3, 12)
```

to 

```
LocalDate(2021, 3, 12)
```
*Advantage* - Consistency in creational patterns with no regard to either Java or Scala. Everything looks like a _smart constructor_
*Disadvantage* - I assume there would be compiler complexity in scanning.

**Comment by Iltotore** [comment]
If I remember correctly, Java's apply is not treated as Scala's apply either.

If I'm right, I would suggest to add Java's apply to this suggestion.
